Packs/GoogleVertexAI/Integrations/GoogleVertexAI/GoogleVertexAI.py

```diff
@@ -0,0 +1,308 @@
+    # https://developers.google.com/identity/protocols/oauth2#:~:text=Refresh%20token%20expiration,-You%20must%20write&text=A%20refresh%20token%20might%20stop,been%20used%20for%20six%20months.
+
+    integration_context: dict = get_integration_context()
+    refresh_token: str = integration_context.get('refresh_token', '')
+
+    data: dict = {
+        'refresh_token': refresh_token,
+        'client_id': CLIENT_ID,
+        'client_secret': CLIENT_SECRET,
+        'grant_type': 'refresh_token'
+    }
+
+    response: requests.Response = requests.post(
+        ACCESS_TOKEN_URL,
+        data=data,
+        verify=DISABLE_SSL
+    )
+
+    if not response.ok:
+        error = error_parser(response)
+        raise ValueError(f'Failed to get refresh token [{response.status_code}] - {error}')
+
+    response_json: dict = response.json()
+    access_token = response_json.get('access_token', '')
+    expires_in: int = response_json.get('expires_in', 3595)
+    refresh_token = response_json.get('refresh_token', '')
+
+    time_now: int = epoch_seconds()
+    time_buffer = 5  # seconds by which to shorten the validity period
+    if expires_in - time_buffer > 0:
+        expires_in -= time_buffer
+    integration_context['refresh_token'] = refresh_token
+    integration_context['access_token'] = access_token
+    integration_context['valid_until'] = time_now + expires_in
+    set_integration_context(integration_context)
+
+    return access_token
+
+
+def epoch_seconds(d: datetime = None) -> int:
+    """
+    Return the number of seconds for given date. If no date, return current.
+    :param d: timestamp datetime object
+    :return: timestamp in epoch
+    """
+    if not d:
+        d = datetime.utcnow()
+    return int((d - datetime.utcfromtimestamp(0)).total_seconds())
+
+
+def resetIntegrationContext():
+    integration_context: dict = get_integration_context()
+    integration_context['refresh_token'] = ""
+    integration_context['access_token'] = ""
+    integration_context['valid_until'] = ""
+    set_integration_context(integration_context)
+    return True
+
+
+def error_parser(resp_err: requests.Response) -> str:
+    """
+    Parse Error
+    """
+    try:
+        response: dict = resp_err.json()
+        if "Unauthorized" in response.get('error_description', ''):
+            resetIntegrationContext()
+            raise ValueError(
+                "The authentication code has been reset. Please reset integration cache for Vetex AI Instance in XSOAR and regenerate the 'Authorization code'")
+        elif "invalid authentication credentials" in response.get('error_description', ''):
+            resetIntegrationContext()
+            raise ValueError(
+                "The authentication code has been reset. Please reset integration cache for Vetex AI Instance in XSOAR and regenerate the 'Authorization code'")
+        elif "Bad" in response.get('error_description', ''):
```

Comment:
If no case is met ?
Please consider adding a general else: case 
With a general Error or an unknown error message

Reply:
Thank you @thefrieddan1 for your feedback all applied. I wrapped the lines after elif in one else block.